### DISCUSSION TITLE
Is there a way to make Yabai remember what windows I've floated/unfloated?

### DISCUSSION THREAD

**Discussion by leftbones** [asmvik/yabai#809] [Q&A] [answered]
I have added a few exceptions for apps like Finder or Email so that they float automatically when opened, but sometimes I like to have them snap in with the rest of the windows. I have a shortcut to do that which is very helpful, but if I float a window, then close it and reopen, it snaps again automatically.

Is there a way to make Yabai remember which windows I've floated/unfloated? Would be SUPER helpful for things like the FireFox dialog window that asks if you want to save or open, since I can't figure out how to add an exception for that one and some others.

Thanks!

**Comment by joshtwo** [comment, marked as answer]
I'd create a hotkey in skhd which will not just float the currently selected window, but add a signal to `window_created` for that same app so that it automatically does so next time. You may want another hotkey which will remove the signal for you if you're tired of floating those windows, or you could even have the hotkey itself look up your "float always signal" for the app to decide whether to remove it or add a new one.

I got this working for me. Thanks for the config idea.
```
# remember float setting
fn - t : app=$(yabai -m query --windows --window mouse | jq -r '.app' | tr ' ' '_');\
         yabai -m window --toggle float;\
         yabai -m signal --remove autofloat_$app || yabai -m signal --add event=window_created app=$app label=autofloat_$app action="yabai -m window $YABAI_WINDOW_ID --toggle float"
```

**Reply by joshtwo** [reply]
Using && with the float toggle will keep this from accidentally adding a new signal when you aren't focused on any windows. Otherwise you might add garbage under `autofloat_`.

```
# remember float setting
fn - t : app=$(yabai -m query --windows --window mouse | jq -r '.app' | tr ' ' '_');\
         yabai -m window --toggle float &&\
         yabai -m signal --remove autofloat_$app || yabai -m signal --add event=window_created app=$app label=autofloat_$app action="yabai -m window $YABAI_WINDOW_ID --toggle float"
```

**Reply by leftbones** [reply]
Thanks!

**Reply by leftbones** [reply]
@joshtwo Hey, I know it's been like two weeks, but I finally got around to adding this to my .skhdrc, and I can't seem to get it to work. Doing the hotkey _does_ float the selected window, but then closing and reopening said window makes it default to it's non-floated state. Am I using it wrong?

Running Big Sur 11.2 on the M1 chip.

**Reply by bbraunj** [reply]
@evprkr @joshtwo I am also having trouble getting this to work on Catalina, SIP disabled.
